handler/proxy.go

```diff
@@ -379,18 +379,20 @@ func (p *Proxy) SetRoundtripContext(req *http.Request, beresp *http.Response) {
 	evalCtx := eval.NewHTTPContext(p.evalContext, p.bufferOption, req, bereq, beresp)
 
 	// Remove blacklisted headers after evaluation to be accessible within our context configuration.
-	if attrCtx == attrReqHeaders {
+	if attrCtx[0] == attrReqHeaders {
 		for _, key := range headerBlacklist {
 			headerCtx.Del(key)
 		}
 	}
 
-	for _, ctxBody := range p.options.Context {
-		options, err := NewCtxOptions(attrCtx, evalCtx, ctxBody)
-		if err != nil {
-			p.log.WithField("parse config", p.String()).Error(err)
+	for _, ctx := range attrCtx {
+		for _, ctxBody := range p.options.Context {
+			options, err := NewCtxOptions(ctx, evalCtx, ctxBody)
```

Comment:
We should add a deprecated log if `options` is not empty for related `ctx`. This should be a warning with a hint which option Couper prefers now.

Reply:
can we do that during parsing/startup? on every single request, that would be too much.

Reply:
No, we cannot. First, it's a temporary solution until we stop supporting `re*_headers`. Secondly, we would lose the flexibility of the request context ala `x-foo = req.xxx`

Reply:
Hm why we lose anything from this feature with a log? This is indeed possible on startup, checking the backend attributes for the outdated names should do the trick.